builder/builder.go

```diff
@@ -378,11 +411,18 @@ func (b *Builder) runBuildingJob(slotCtx context.Context, proposerPubkey boostTy
 		queueMu.Unlock()
 	}
 
+	// Avoid submitting early into a given slot. For example if slots have 12 second interval, submissions should
+	// not begin until 8 seconds into the slot.
+	slotTime := time.Unix(int64(attrs.Timestamp), 0).UTC()
+	submitTime := slotTime.Add(-SubmissionDelaySecondsDefault)
+
+	ctx = context.WithValue(ctx, key("slot"), attrs.Slot)
+	ctx = context.WithValue(ctx, key("timestamp"), uint64(attrs.Timestamp))
 	// Empties queue, submits the best block for current job with rate limit (global for all jobs)
-	go runResubmitLoop(ctx, b.limiter, queueSignal, submitBestBlock)
+	go runResubmitLoop(ctx, b.limiter, queueSignal, submitBestBlock, &submitTime)
```

Comment:
We pass in some of the values as arguments, and some as context values - any reason not to pass in everything as normal values?

Reply:
I've removed the context values because that was meant for investigating a separate issue unrelated to this PR. To reduce scope I'll add those changes to a separate branch, so resolving this